src/dom.ts

```diff
@@ -229,44 +227,87 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
     return point;
   }
 
-  async _performPointerAction(action: (point: types.Point) => Promise<void>, options: PointerActionOptions & types.PointerActionWaitOptions & types.NavigatingActionWaitOptions = {}): Promise<void> {
+  async _retryPointerAction(action: (point: types.Point) => Promise<void>, options: PointerActionOptions & types.PointerActionWaitOptions & types.NavigatingActionWaitOptions = {}): Promise<void> {
     const deadline = this._page._timeoutSettings.computeDeadline(options);
-    const { force = false } = (options || {});
-    if (!force)
+    while (!helper.isPastDeadline(deadline)) {
+      const result = await this._performPointerAction(action, deadline, options);
+      if (result === 'done')
+        return;
+    }
+    throw new TimeoutError(`waiting for element to receive pointer events failed: timeout exceeded`);
+  }
+
+  async _performPointerAction(action: (point: types.Point) => Promise<void>, deadline: number, options: PointerActionOptions & types.PointerActionWaitOptions & types.NavigatingActionWaitOptions = {}): Promise<'done' | 'retry'> {
+    const { force = false, position } = options;
+    if (!force && !(options as any).__testHookSkipStablePosition)
       await this._waitForDisplayedAtStablePosition(deadline);
-    const position = options ? options.position : undefined;
+
+    if ((options as any).__testHookAfterStablePosition)
+      await (options as any).__testHookAfterStablePosition();
+
+    this._page._log(inputLog, 'triggering hover effects...');
     await this._scrollRectIntoViewIfNeeded(position ? { x: position.x, y: position.y, width: 0, height: 0 } : undefined);
-    const point = position ? await this._offsetPoint(position) : await this._clickablePoint();
-    point.x = (point.x * 100 | 0) / 100;
-    point.y = (point.y * 100 | 0) / 100;
-    await this._page.mouse.move(point.x, point.y);  // Force any hover effects before waiting for hit target.
-    if (options && (options as any).__testHookBeforeWaitForHitTarget)
-      await (options as any).__testHookBeforeWaitForHitTarget();
-    if (!force)
-      await this._waitForHitTargetAt(point, deadline);
+    let point = roundPoint(position ? await this._offsetPoint(position) : await this._clickablePoint());
+    await this._page.mouse.move(point.x, point.y);  // Force any hover effects.
```

Comment:
This should happen after the pause (and after the second scroll into view)

Reply:
This would cut off all `onmouseover`, `onscroll` etc handlers. Let's start with that but I feel like tested page is going to diverge too far from the original (unpaused).